libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6/hwdef.dat

```diff
@@ -0,0 +1,231 @@
+# hw definition file for processing by chibios_hwdef.py for A6_YJUAV
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# crystal frequency
+OSCILLATOR_HZ 16000000
+
+# board ID for firmware load
+APJ_BOARD_ID 1113
+
+FLASH_SIZE_KB 2048
+
+# with 2M flash we can afford to optimize for speed
+env OPTIMIZE -O2
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 5
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART5 UART8 UART7 OTG2
+
+# now we define the pins that USB is connected on
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# USART1 TELEM1
+PA10 USART1_RX USART1
+PA9 USART1_TX USART1
+
+# USART2 TELEM2
+PD6 USART2_RX USART2
+PD5 USART2_TX USART2
+
+# USART3 GPS
+PD9 USART3_RX USART3 NODMA
+PD8 USART3_TX USART3 NODMA
+
+# UART5 GPS2
+PB5 UART5_RX UART5 NODMA
+PB13 UART5_TX UART5 NODMA
+
+# SBUS, DSM port
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# UART7 DEBUG
+PE7 UART7_RX UART7 NODMA
+PE8 UART7_TX UART7 NODMA
+
+# default the 2nd interface to MAVLink2 until MissionPlanner updates drivers
+define HAL_OTG2_PROTOCOL SerialProtocol_MAVLink2
+
+# these are the pins for SWD debugging with a STlinkv2 or black-magic probe
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SPI1 -IMU1 -COMPASS
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PD7 SPI1_MOSI SPI1
+
+# SPI2 -IMU2 -IMU3
+PD3 SPI2_SCK SPI2
+PC2 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+# SPI4 -FRAM -BAROMETER
+PE2 SPI4_SCK SPI4
+PE5 SPI4_MISO SPI4
+PE6 SPI4_MOSI SPI4
+
+# SPI3 -Extra SPI
+PB3 SPI3_SCK SPI3
+PB4 SPI3_MISO SPI3
+PB2 SPI3_MOSI SPI3
+
+# sensors cs
+PA8  IMU1_CS CS
+PD4  IMU2_CS CS
+PD11 IMU3_CS CS
+PE4  FRAM_CS CS
+PE3  BAROMETER_CS CS
+PD10 COMPASS_CS CS
+PC15 RESERVE_CS CS
+
+# Extra SPI CS
+PE10 EXT_CS CS
+
+# I2C buses
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+PD12 I2C4_SCL I2C4
+PD13 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C2 I2C4
+
+define HAL_I2C_INTERNAL_MASK 0
+
+# PWM channels
+PE9  TIM1_CH1  TIM1 PWM(1)  GPIO(50)
```

Comment:
I recommend you add bi-directional dshot to the first 8 channels. Since you are using H743 this is a very easy change. You can add "BIDIR" on the end of PWM 1, 3, 5, 7 and that will give you 8 bi-directional channels (assuming their are no level shifters on the outputs?)

Reply:
![img_v2_eedcb7aa-314f-43ea-916c-19a0eb7f00fg](https://github.com/ArduPilot/ardupilot/assets/116018265/14fd9604-7355-4bef-ab73-071dad5e9990)